telegram/ext/_baseupdateprocessor.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+#
+# A library that provides a Python interface to the Telegram Bot API
+# Copyright (C) 2015-2023
+# Leandro Toledo de Souza <devs@python-telegram-bot.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains the BaseProcessor class."""
+from abc import abstractmethod
+from asyncio import BoundedSemaphore
+from types import TracebackType
+from typing import Any, Awaitable, Optional, Type
+
+
+class BaseUpdateProcessor:
```

Comment:
```suggestion
class BaseUpdateProcessor(ABC):
```
TBH I'm astonished that this even worked without … see my question at https://stackoverflow.com/questions/76292008